### PR TITLE
chore: type centroid util

### PR DESCRIPTION
**Pull request by seve**
Type the centroid util and call sites as well as needed imports.

Note:

I did some retyping of the `CategeoricalColumnSchema` to make the categories field required on columns with type `categorical'.  We'll have to do some tuning (#35) to reduce casting in favor of type guarding.

I also had to untype `nextSharedState` (it was typing as RootState) as I believe it opens up a self reference .  This should be fixed in #9  cc: @colinmegill

### CODE REVIEW

File: client/src/util/stateManager/schemaHelpers.ts [hunk 1]
```diff
@@ -124,16 +132,15 @@ export function removeObsAnnoCategory(
 export function addObsAnnoCategory(schema: any, name: any, category: any) {
   /* add a category to a categorical annotation */
   const categories = schema.annotations.obsByName[name]?.categories;
-  if (!categories)
```
Comment (tihuan) [inline client/src/util/stateManager/schemaHelpers.ts:127]: ah are we moving this runtime check because `categories` is guaranteed somehow?

**Review by tihuan — APPROVED** [review]
LGTM!! Thanks for updating the types, @seve 🙏 Just one question! Thank you!

File: client/src/util/centroid.ts [hunk 2]
```diff
@@ -16,22 +21,22 @@ import {
...
-  schema: any,
+  schema: Schema,
   categoryName: string,
   categoryDf: Dataframe,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  layoutChoice: any,
+  layoutChoice: LayoutChoiceState,
   layoutDf: Dataframe
-) => {
+): Map<string, CentroidCoordinateObject> => {
```
Comment (bkmartinjr) [inline client/src/util/centroid.ts:39]: we also support non-string labels.   Shouldn't this be typed:   `Map<LabelType, CentroidCoordinateObject>`?

File: client/src/util/centroid.ts [hunk 3]
```diff
@@ -105,14 +111,12 @@ const getCoordinatesByLabel = (
...
-  schema: any,
+  schema: Schema,
   categoryName: string,
   categoryDf: Dataframe,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  layoutChoice: any,
+  layoutChoice: LayoutChoiceState,
   layoutDf: Dataframe
-) => {
+): Map<string, [number, number]> => {
```
Comment (bkmartinjr) [inline client/src/util/centroid.ts:119]: See above comment about assuming all labels are strings.

**Review by bkmartinjr — CHANGES_REQUESTED** [review]
Overall looks good, with one caveat: the centrol helpers assume all labels are strings, which is incorrect AFAIK.  We can have datasets with non-string categoricals (numbers, bools, etc).

File: client/src/util/centroid.ts (hunk 2 above)
Comment (seve) [reply client/src/util/centroid.ts:39]: I wonder why I didn't get a type error for that, is it just because all of those types can be cast to string?

Comment (bkmartinjr) [reply client/src/util/centroid.ts:39]: you have it right.  A string is a LabelType, but the converse is not true.

Typing only helps you in some cases ;)

File: client/__tests__/util/centroid.test.ts [hunk 4]
```diff
@@ -31,7 +31,11 @@ describe("centroid", () => {
       schema,
       "field4",
       obsAnnotations,
```
Comment (bkmartinjr) [inline client/__tests__/util/centroid.test.ts:33]: Optional:  given the confusion around the type of an obsAnnnotation, you may want to add tests which exercise the other label types (eg, number or boolean or blended)

**Comment by bkmartinjr** [comment]
@seve - you re-requested a review, but I don't see any changes to `util/centroid.ts` -- it still assumes that all obs categories contain only strings, as far as I can tell...

**Review by bkmartinjr — APPROVED** [review]
LGTM.

You might consider additional test coverage as noted inline.

**Pull request merged**